src/validators/ArrayValidator.ts

```diff
@@ -95,6 +110,26 @@ export class ArrayValidator<T> extends BaseValidator<T[]> {
 			? Result.ok(transformed)
 			: Result.err(new CombinedPropertyError(errors));
 	}
+
+	private handleUniqueStrategy(givenValues: unknown[]) {
+		if (givenValues.length < 2) return [];
+		const errors: [number, BaseError][] = [];
+		const values = [...givenValues];
+		let previousValue = values.shift();
+
+		for (let i = 0; i < values.length; i++) {
+			const value = values[i];
+			if (!Object.is(previousValue, value))
```

Comment:
Did you mean to negate this?

Reply:
See also comment below about using Set instead of this loop